### PR TITLE
Use Marathon 1.1.1

### PR DESCRIPTION
**Pull request by aquamatthias**

### CODE REVIEW

**Comment by sttts** [comment]
bildschirmfoto 2016-04-18 um 11 11 21

**Comment by sttts** [comment]
After clicking "Edit Configuration" and entering an Id (that field is empty but probably shoulnd't) install works.

**Comment by sttts** [comment]
Installation with the cli works.

**Comment by aquamatthias** [comment]
@stts added 2 parameters set by default:
--default_accepted_resource_roles *
--mesos_role <framework name>

File: repo/packages/M/marathon/2/marathon.json.mustache [hunk 1]
```diff
@@ -36,7 +36,7 @@
...
...#marathon.task-launch-timeout}}--task_launch_timeout {{marathon.task-launch-timeout}}{{/marathon.task-launch-timeout}} {{#marathon.task-reservation-timeout}}--task_reservation_timeout {{marathon.task-reservation-timeout}}{{/marathon.task-reservation-timeout}} {{#marathon.tracing}} --tracing{{/marathon.tracing}}{{^marathon.tracing}}--disable_tracing{{/marathon.tracing}} {{#marathon.webui-url}}--webui_url {{marathon.webui-url}}{{/marathon.webui-url}} --zk {{#marathon.zk}}{{marathon.zk}}{{/marathon.zk}}{{^marathon.zk}}zk://master.mesos:2181/universe/{{marathon.framework-name}}{{/marathon.zk}} {{#marathon.zk-compression}} --zk_compression{{/marathon.zk-compression}}{{^marathon.zk-compression}}--disable_zk_compression{{/marathon.zk-compression}} {{#marathon.zk-compression-threshold}}--zk_compression_threshold {{marathon.zk-compression-threshold}}{{/marathon.zk-compression-threshold}} {{#marathon.zk-max-versions}}--zk_max_versions {{marathon.zk-max-versions}}{{/marathon.zk-max-versions}} {{#marathon.zk-session-timeout}}--zk_session_timeout {{marathon.zk-session-timeout}}{{/marathon.zk-session-timeout}} {{#marathon.zk-timeout}}--zk_timeout {{marathon.zk-timeout}}{{/marathon.zk-timeout}}",
```
Comment (sttts) [inline repo/packages/M/marathon/2/marathon.json.mustache]: That's a real oneliner :)

Comment (aquamatthias) [inline repo/packages/M/marathon/2/marathon.json.mustache]: Nearly impossible to review I know - sorry.

File: repo/packages/M/marathon/2/package.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "scm" : "https://github.com/mesosphere/marathon.git",
+  "maintainer" : "support@mesosphere.io",
+  "description" : "A container orchestration platform for Mesos and DCOS.",
+  "framework" : true,
+  "licenses" : [ {
+    "name" : "Apache License Version 2.0",
+    "url" : "https://github.com/mesosphere/marathon/blob/master/LICENSE"
+  } ],
+  "tags" : [ "init", "long-running" ],
+  "preInstallNotes" : " We recommend a minimum of one node with at least 2 CPU shares and 1GB of RAM available for the Marathon DCOS Service.",
```
Comment (sttts) [inline repo/packages/M/marathon/2/package.json]: The leading space is intended?

File: repo/packages/M/marathon/2/marathon.json.mustache [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
```
Comment (sttts) [inline repo/packages/M/marathon/2/marathon.json.mustache]: Why no default value for `default-accepted-resource-roles` in `config.json`?

File: repo/packages/M/marathon/2/config.json [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+        "max-apps" : {
+          "description" : "The maximum number of applications that may be created.",
+          "type" : "integer"
+        },
+        "max-tasks-per-offer" : {
+          "description" : "Maximum tasks per offer. Do not start more than this number of tasks on a single offer.",
+          "type" : "integer",
+          "default" : 1
+        },
+        "mesos-authentication-principal" : {
```
Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: This defaults to the framework name. This should be part of the description.

File: repo/packages/M/marathon/2/config.json [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+          "description" : "Mesos Authentication Secret.",
+          "type" : "string"
+        },
+        "mesos-leader-ui-url" : {
+          "description" : "The host and port (e.g. \"http://mesos_host:5050\") of the Mesos master.",
+          "type" : "string",
+          "default" : "/mesos"
+        },
+        "mesos-role" : {
+          "description" : "Mesos role for this framework. If set, Marathon receives resource offers for the specified role in addition to resources with the role designation '*'.",
```
Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: Defaults to the framework name.

**Comment by sttts** [comment]
@aquamatthias added a few comments. Other than that I am still testing health checks. I am experiencing some strange behavior. Still investigating.

**Comment by sttts** [comment]
Health check issue was a false positive. Seems to work fine.

File: repo/packages/M/marathon/2/package.json (hunk 2 above)
Comment (aquamatthias) [inline repo/packages/M/marathon/2/package.json]: No, will remove it.

File: repo/packages/M/marathon/2/marathon.json.mustache (hunk 3 above)
Comment (aquamatthias) [inline repo/packages/M/marathon/2/marathon.json.mustache]: Good suggestion. Will move.

File: repo/packages/M/marathon/2/config.json (hunk 4 above)
Comment (aquamatthias) [inline repo/packages/M/marathon/2/config.json]: I can not set a default value here.
Or do you think this should be part of the description?

File: repo/packages/M/marathon/2/config.json (hunk 5 above)
Comment (aquamatthias) [inline repo/packages/M/marathon/2/config.json]: I can not set a default value here.
Or do you think this should be part of the description?

Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: Yes, the later.

File: repo/packages/M/marathon/2/config.json (hunk 4 above)
Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: Something like `(default: framework name)` is fine.

**Comment by aquamatthias** [comment]
@sttts addressed your comments.

**Comment by sttts** [comment]
Looks good to me. 

The only open issue are persistent volumes? Is there any relation to this PR?

**Comment by aquamatthias** [comment]
No this problem is dcos specific.

File: repo/packages/M/marathon/2/config.json [hunk 6]
```diff
@@ -0,0 +1,392 @@
+{
+  "additionalProperties" : false,
+  "description" : "Marathon DCOS Service properties",
+  "properties" : {
+    "application" : {
```
Comment (mlunoe) [inline repo/packages/M/marathon/2/config.json]: I think this should be called `service` like the other packages does

File: repo/packages/M/marathon/2/config.json [hunk 7]
```diff
@@ -0,0 +1,392 @@
...
+      "additionalProperties" : false,
+      "description" : "Marathon app configuration properties.",
+      "properties" : {
+        "id" : {
```
Comment (mlunoe) [inline repo/packages/M/marathon/2/config.json]: And this `name`, again as the other packages does it

Comment (mlunoe) [inline repo/packages/M/marathon/2/config.json]: See here: https://github.com/mesosphere/universe/blob/1-7ea/repo/packages/K/kafka/6/config.json#L7

**Comment by aquamatthias** [comment]
@sttts updated this PR according to mlunoe's suggestions (naming conventions).
Tested it with 1.7 successfully.

File: repo/packages/M/marathon/2/config.json [hunk 8]
```diff
@@ -89,7 +89,8 @@
         },
         "default-accepted-resource-roles" : {
           "description" : "Default for the defaultAcceptedResourceRoles attribute of all app definitions as a comma-separated list of strings. This defaults to all roles for which this Marathon instance is configured to receive offers.",
```
Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: Nit: "Default for the *default*...Roles attribute" sounds strange.

File: repo/packages/M/marathon/2/config.json [hunk 9]
```diff
@@ -127,8 +129,7 @@
         },
         "framework-name" : {
```
Comment (sttts) [inline repo/packages/M/marathon/2/config.json]: This is not necessary anymore.

File: repo/packages/M/marathon/2/marathon.json.mustache [hunk 10]
```diff
@@ -36,9 +36,9 @@
...
   "labels": {
-    "DCOS_PACKAGE_FRAMEWORK_NAME": "{{marathon.framework-name}}"
+    "DCOS_PACKAGE_FRAMEWORK_NAME": "{{#marathon.framework-name}}{{marathon.framework-name}}{{/marathon.framework-name}}{{^marathon.framework-name}}{{service.name}}{{/marathon.framework-name}}"
```
Comment (sttts) [inline repo/packages/M/marathon/2/marathon.json.mustache]: s/marathon.framework-name/service.name/g

**Pull request merged**